internal/pkg/create/create.go

```diff
@@ -520,18 +522,17 @@ func InstallProjectDependencies(
 	return outputs
 }
 
-// generateGitZipFileURL will return template's GitHub zip file download link
-// In the future, this function can be extended to support other Git hosts, such as GitLab.
-// TODO, @cchensh, we should get prepared for other non-Git hosts and refactor the create pkg
-func generateGitZipFileURL(templateURL string, gitBranch string) string {
+// generateGitZipFileURL will return the GitHub zip URL for a templateURL.
+func generateGitZipFileURL(httpClient slackdeps.HTTPClient, templateURL string, gitBranch string) string {
 	zipURL := strings.Replace(templateURL, ".git", "", -1) + "/archive/refs/heads/"
 
 	if gitBranch == "" {
 		mainURL := zipURL + "main.zip"
 		masterURL := zipURL + "master.zip"
-		zipURL = deputil.URLChecker(mainURL)
+
+		zipURL = deputil.URLChecker(httpClient, mainURL)
 		if zipURL == "" {
-			zipURL = deputil.URLChecker(masterURL)
+			zipURL = deputil.URLChecker(httpClient, masterURL)
 		}
 	} else {
 		zipURL = zipURL + gitBranch + ".zip"
```

Comment:
note: Kinda funny that we never apply `URLChecker` to custom branches. Work for the future!